server/@antlr4-tsplang/test/tspLexer.test.ts

```diff
@@ -0,0 +1,203 @@
+                        tsp1: false,
+                    },
+                    {
+                        name: `Tokenizes the ">>" (bitwise right shift) operator`,
+                        content: "x >> y",
+                        tokenNames: ["NAME", "HORIZONTAL_WS", "BIT_RS", "HORIZONTAL_WS", "NAME"],
+                        tsp1: false,
+                    },
+                    {
+                        name: `Tokenizes the "!" (logical NOT) operator`,
+                        content: "!x",
+                        tokenNames: ["LOGICAL_NOT", "NAME"],
+                        tsp1: false,
+                    },
+                    {
+                        name: `Tokenizes boolean "true"`,
+                        content: "true",
+                        tokenNames: ["BOOLEAN"],
+                        tsp1: true,
+                    },
+                    {
+                        name: `Tokenizes boolean "false"`,
+                        content: "false",
+                        tokenNames: ["BOOLEAN"],
+                        tsp1: true,
+                    },
+                    {
+                        name: `Tokenizes "local" keyword`,
+                        content: "local x",
+                        tokenNames: ["LOCAL", "HORIZONTAL_WS", "NAME"],
+                        tsp1: true,
+                    },
+                    {
+                        name: `Tokenizes "..." (vararg) keyword`,
+                        content: "...",
+                        tokenNames: ["VARARG"],
+                        tsp1: true,
+                    },
+                    {
+                        name: `Tokenizes long strings (surrounded by "[[...]]")`,
+                        content: `[[String with
+                        some newlines and
+                        such]]`,
+                        tokenNames: ["LONGSTRING"],
+                        tsp1: true,
+                    },
+                    {
+                        name: `Tokenizes long commments (surrounded by "--[[...]]")`,
+                        content: `--[[String with
+                        some newlines and
+                        such]]`,
+                        tokenNames: ["LONGCOMMENT"],
+                        tsp1: true,
+                    },
+                    {
+                        name: `Tokenizes line commments (started by "--...")`,
+                        content: `--Single line comment
+                        `,
+                        tokenNames: ["LINE_COMMENT", "HORIZONTAL_WS"],
+                        tsp1: true,
+                    },
+                    {
+                        name: `Tokenizes shebang line (started by "#!")`,
+                        content: `#!shebang text`,
+                        tokenNames: ["SHEBANG"],
+                        tsp1: true,
+                    },
+                ]
```

Comment:
Quote precedence is double (`"`) -> single (`'`) -> template (`` ` ``).
Change the `name` property to follow quote precedence.